lib/Common/Memory/Allocator.h

```diff
@@ -382,16 +384,19 @@ void DestructArray(size_t count, T* obj)
 template <typename TAllocator, typename T>
 void DeleteArray(typename AllocatorInfo<TAllocator, T>::AllocatorType * allocator, size_t count, T * obj)
 {
-    if (count == 0)
+    if (count == 0 && TAllocator::FakeZeroLengthArray)
```

Comment:
AllocatorInfo<TAllocator, T>::AllocatorType is just same as TAllocator...

we care because for Heap allocated zero length array, we still allocate memory for the tracking data. and for Heap allocator the tracking data is in front of the user address, need to call into the free function to clear the tracking data and reset the memory leaking status